### ISSUE TITLE
Gem::Version raising ArgumentError for weirdly versioned Jenkins plugins

### ISSUE THREAD

**Issue by ampledata** [opscode-cookbooks/jenkins#292]
[Gem::Version](http://ruby-doc.org/stdlib-2.0.0/libdoc/rubygems/rdoc/Gem/Version.html), introduced in @schisamo's epic 41bc1540868781f721396815db36010d45185bda commit, is rejecting the version number for the '[GitHub pull request builder plugin](https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin)' (ghprb) of '1.16-5' as malformed, leading to the raising of an ArgumentError exception and subsequent failure to converge.

To Reproduce
---
1. Create a Recipe as follows:

    ```
    include_recipe 'jenkins::server'
    
    jenkins_plugin 'ghprb' do
      version '1.16-5'
    end
    ```

2. Converge (chef-client, chef-solo, w/e).

Expected Result
---
1. Jenkins should be installed in a server role.
2. Version '1.16-5' of the ghprb Plugin should be installed.

Actual Result
---
1. Jenkins is installed in a server role.
2. Version '1.16-5' of the ghprb Plugin is not installed.
3. Chef fails to converge with an ArgumentError Exception.

Work Around
---
Unknown at this time.

Affects Versions
---
1. Jenkins Cookbook 2.2.1
2. GitHub pull request builder plugin 1.16-5

Logs
---

```
================================================================================

    Error executing action `install` on resource 'jenkins_plugin[ghprb]'

================================================================================       
           
    ArgumentError       
    -------------       
    Malformed version number string 1.16-5       
           
    Cookbook Trace:       
    ---------------       
           /tmp/kitchen/cookbooks/jenkins/libraries/plugin.rb:258:in `new'
           /tmp/kitchen/cookbooks/jenkins/libraries/plugin.rb:258:in `install_plugin_from_update_center'
           /tmp/kitchen/cookbooks/jenkins/libraries/plugin.rb:124:in `block (2 levels) in <class:JenkinsPlugin>'
           /tmp/kitchen/cookbooks/jenkins/libraries/plugin.rb:155:in `block in <class:JenkinsPlugin>'
       
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cookbooks/obci/recipes/jenkins_292.rb
           
             4: jenkins_plugin 'ghprb' do
             5:   version :latest
       
             6: end
           
           Compiled Resource:
       
           ------------------
           
       # Declared in /tmp/kitchen/cookbooks/obci/recipes/jenkins_292.rb:4:in `from_file'
           
       
           jenkins_plugin("ghprb") do
       
             action :install
             retries 0
             retry_delay 2
             guard_interpreter :default
             cookbook_name :obci
             recipe_name "jenkins_292"
             version :latest
             install_deps true
           end
       
           
       [2014-12-04T00:50:39+00:00] INFO: Running queued delayed notifications before re-raising exception
       
       
```

**Comment by ericcitaire** [comment]
Same issue with build-monitor-plugin : 
Malformed version number string 1.6+build.135

**Comment by schisamo** [comment]
This was fixed in 1e9e591.
